application/libraries/SessionData.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace LimeSurvey;
+
+use CHttpSession;
+use ArrayAccess;
+
+/**
+ * Session Data
+ *
+ * Simple array access wrapper of CHttpSession to allow
```

Comment:
Can you explain again why this is needed instead of just using CHttpSession? Purism reason or pragmatic? :))

Reply:
I cant mock in the data in CHttpSession.

Reply:
Why not? Would you just do

```php
$mockSession = $this
  ->createMock(CHttpSession::class)
  ->method('getX')
  ->willReturn($yourData);
```
Or similar?

Reply:
If it was that simple I would have done that. I tried to figure out a way to mock it and it looked to be super messy so I went with a simpler approach.

Reply:
Could you explain why you couldn't do it like that? I don't get it.

Reply:
Cool cool, will resolve this.